drivers/d3d12/rendering_shader_container_d3d12.cpp

```diff
@@ -72,6 +72,97 @@ GODOT_GCC_WARNING_POP
 GODOT_CLANG_WARNING_POP
 GODOT_MSVC_WARNING_POP
 
+// SPIR-V to DXIL does way too many allocations, which causes worker thread pools
+// to bottleneck each other due to sharing the same global process heap.
+// This can be solved by making each thread allocate from its own heap.
+#define SPIRV_TO_DXIL_ENABLE_HEAP_PER_THREAD
+
+#ifdef SPIRV_TO_DXIL_ENABLE_HEAP_PER_THREAD
+
+namespace {
+struct Win32Heap {
+	HANDLE handle;
+	SafeRefCount ref_count;
+
+	Win32Heap() {
+		handle = HeapCreate(0, 0, 0);
```

Comment:
Shouldn't this use Godot's memory allocation functions instead of using the Windows API directly?

Or is there a reason to specifically create a heap here?

My gut feeling is that we should be using our own memory allocation functions instead of creating and destroying heaps manually

CC @Ivorforce

Reply:
I think there's no direct generic C++ equivalent to this (would likely involve some form of `mmap` manipulations in case on Unix), and we probably do not want to write a completely custom memory allocator just for this, so it seems reasonable to use OS API.

But since it's used by Mesa-NIR, should not it be added directly to the https://github.com/godotengine/godot-nir-static/ instead?